src/query/api/v1/handler/prometheus/remote/read.go

```diff
@@ -202,22 +207,31 @@ func (h *PromReadHandler) read(
 			cancelFuncs[i] = cancel
 			query, err := storage.PromReadQueryToM3(promQuery)
 			if err != nil {
-				multiErrLock.Lock()
+				mu.Lock()
 				multiErr = multiErr.Add(err)
-				multiErrLock.Unlock()
+				mu.Unlock()
 				return
 			}
 
 			// Detect clients closing connections
 			handler.CloseWatcher(ctx, cancel, w, h.instrumentOpts)
 			result, err := h.engine.Execute(ctx, query, queryOpts, fetchOpts)
 			if err != nil {
-				multiErrLock.Lock()
+				mu.Lock()
 				multiErr = multiErr.Add(err)
-				multiErrLock.Unlock()
+				mu.Unlock()
 				return
 			}
 
+			mu.Lock()
+			if !exhaustiveSet {
```

Comment:
Do you need to have this second var `exhaustiveSet` if you just set `exhaustive = true` at the beginning then always do `exhaustive = exhaustive && result.Exhaustive`? 

It should always be false if any `result.Exhaustive` is actually false and will default to true if for some reason you have zero queries.

Reply:
Yeah good call, much nicer this way